GoogleForCreators/web-stories-wp
# Text edit mode + Moveable resizing/rotation

## Summary

Adds `SingleSelectionMoveable` to Text edit mode, reuses the same logic mostly with some special handling.


## Relevant Technical Choices
- The Moveable/element wrapper needs to be visually updated when the text is changing the height of the element while being in edit mode.
- If the element height is updated due to doing things in edit mode, and then the user wants to resize the element, the element properties need to be updated before the user can do that -- otherwise, the calculations will be based on the original element.
- Dragging is not allowed in edit mode, however, we can't just disable it since otherwise selecting part of the text will be disabled.


## To-do




## User-facing changes

The user can now resize and rotate the text element while in edit mode.


## Testing Instructions

I
1. Add a Text element and enter edit mode.
2. Verify that resizing and rotation is enabled in edit mode as well and the element is rotated/resized as expected.

II
1. Add a Text element and enter edit mode.
2. Add some empty lines / new lines and some text to change the element height while in edit mode.
3. Verify that the blue line for resizing/rotation is following the new text height.
4. Now, without exiting the edit mode, resize/rotate the element.
5. Verify that the resizing/rotation is working as expected.


---



Fixes #1801
